pkg/zsio/writer.go

```diff
@@ -84,19 +83,54 @@ func (w *Writer) writeValue(val []byte) error {
 	if val == nil {
 		return w.write("-;")
 	}
-	if err := w.write(zsonEscape(string(val))); err != nil {
+	if err := w.writeEscaped(val); err != nil {
 		return err
 	}
 	return w.write(";")
 }
 
-func zsonEscape(s string) string {
-	if s == "-" {
-		return "\\-"
-	}
+func (w *Writer) escape(c byte) error {
+	return w.write(fmt.Sprintf("\\x%02x", c))
```

Comment:
I think ```fmt.Fprintf(w.writer, fmt, `\x%02x`, c)``` will avoid a call to runtime.slicebytetostring that's hidden in fmt.Sprintf along with a call to runtime.stringtobyteslice that's hidden in Writer.write, but that's definitely marginal stuff.

Reply:
It's on the slow path but ok.  How about this?
```
func (w *Writer) escape(c byte) error {
	const hex = "0123456789abcdef"
	var b [4]byte
	b[0] = '\\'
	b[1] = 'x'
	b[2] = hex[c>>4]
	b[3] = hex[c&0xf]
	_, err := w.writer.Write(b[:])
	return err
}
```

Reply:
👍